lib/SILOptimizer/Analysis/ColdBlockInfo.cpp

```diff
@@ -221,31 +238,99 @@ bool ColdBlockInfo::inferFromEdgeProfile(SILBasicBlock *BB) {
   SmallVector<ProfileCounter, 2> succCount;
 
   // Current analysis only accurately handles blocks with 2 successors,
-  // especially since we only have two temperatures.
+  // especially since we only have two temperatures (cold/warm).
+  // TODO: With propagation algorithm, this limitation can be removed to support
+  // multi-way branches (switch statements) and arbitrary successor counts.
```

Comment:
nit: inaccurate addition to this comment

There already is propagation. The reason for this limitation is hinted at in the original comment: there's only two temperatures being set and propagated for blocks. We set a block cold only if it's taken with very low probability, currently under 3%, otherwise it's just "warm".  If a block has 100 successors and they are all taken with equal probability of 1%, then we'd mistakenly mark all blocks cold.

Definitely could use a better strategy here, but I think for this PR let's stay focused on fixing the divide-by-zero and having the "assume cold for missing data".